partitioned-heat-conduction/solver-fenics/heat_pySDC.py

```diff
@@ -0,0 +1,346 @@
+  u = 1 + x^2 + alpha*y^2 + \beta*t
+  f = beta - 2 - 2*alpha
+
+Heat equation with mixed boundary conditions. (Neumann problem)
+  u'= Laplace(u) + f  in the shifted unit square [1,2] x [0,1]
+  du/dn = f_N         on the coupling boundary at x = 1
+  u = u_D             on the remaining boundary
+  u = u_0             at t = 0
+  u = 1 + x^2 + alpha*y^2 + \beta*t
+  f = beta - 2 - 2*alpha
+"""
+
+from __future__ import print_function, division
+from fenics import Function, FunctionSpace, Expression, Constant, DirichletBC, TrialFunction, TestFunction, \
+    File, solve, grad, inner, dx, interpolate, VectorFunctionSpace, MeshFunction, MPI
+from fenicsprecice import Adapter
+from errorcomputation import compute_errors
+from my_enums import ProblemType, DomainPart
+import argparse
+import numpy as np
+from problem_setup import get_geometry
+import sympy as sp
+
+from pySDC.implementations.sweeper_classes.imex_1st_order_mass import imex_1st_order_mass
+from pySDC.implementations.controller_classes.controller_nonMPI import controller_nonMPI
+from heat_pySDC_problem_class import fenics_heat_2d
+
+def determine_gradient(V_g, u, flux):
+    """
+    compute flux following http://hplgit.github.io/INF5620/doc/pub/fenics_tutorial1.1/tu2.html#tut-poisson-gradu
+    :param V_g: Vector function space
+    :param u: solution where gradient is to be determined
+    :param flux: returns calculated flux into this value
+    """
+
+    w = TrialFunction(V_g)
+    v = TestFunction(V_g)
+
+    a = inner(w, v) * dx
+    L = inner(grad(u), v) * dx
+    solve(a == L, flux)
+
+
+
+parser = argparse.ArgumentParser(description="Solving heat equation for simple or complex interface case")
+parser.add_argument("participantName", help="Name of the solver.", type=str, choices=[p.value for p in ProblemType])
+parser.add_argument("-e", "--error-tol", help="set error tolerance", type=float, default=10**-8,)
+
+args = parser.parse_args()
+participant_name = args.participantName
+
+
+# preCICE error tolerance
+# Error is bounded by coupling accuracy. In theory we would obtain the analytical solution.
+error_tol = args.error_tol
+
+if participant_name == ProblemType.DIRICHLET.value:
+    problem = ProblemType.DIRICHLET
+    domain_part = DomainPart.LEFT
+elif participant_name == ProblemType.NEUMANN.value:
+    problem = ProblemType.NEUMANN
+    domain_part = DomainPart.RIGHT
+    exit("Neumann problem not yet supported with pySDC")
+
+domain_mesh, coupling_boundary, remaining_boundary = get_geometry(domain_part)
+
+# Define function space using mesh
+V = FunctionSpace(domain_mesh, 'P', 2)
+V_g = VectorFunctionSpace(domain_mesh, 'P', 1)
+W = V_g.sub(0).collapse()
+
+
+##################################################################
+# Problem definition and preCICE initialization
+##################################################################
+
+# Time step size
+# Should be integer fraction of the used time window size
+precice_time_window_size = 0.1
```

Comment:
This is something that we should get from the precice config. Please use the following:
```suggestion
precice_time_window_size = precice.get_max_time_step_size()
```
Note: At the beginning of the window the max step size equals the window size.